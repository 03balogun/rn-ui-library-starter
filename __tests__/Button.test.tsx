import {render, screen} from '@testing-library/react-native';
import * as React from 'react';

import {Button} from '../src';

describe('<Button />', () => {
  const mockFn = jest.fn();
  const BUTTON_LABEL = 'Register';

  it('renders correctly', () => {
    const tree = render(
      <Button text={BUTTON_LABEL} onPress={mockFn} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correct text label', () => {
    render(<Button text={BUTTON_LABEL} onPress={mockFn} />);

    const textLabel = screen.getByText(BUTTON_LABEL);
    expect(textLabel.props.children).toBe(BUTTON_LABEL);
  });
});
