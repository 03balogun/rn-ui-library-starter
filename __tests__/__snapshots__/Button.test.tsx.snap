// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Register"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "purple",
        "borderRadius": 4,
        "flexGrow": 0,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          false,
        ]
      }
    >
      Register
    </Text>
  </View>
</View>
`;
